src/main/scala/com/fulcrumgenomics/rnaseq/CollectErccMetrics.scala

```diff
@@ -0,0 +1,294 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.rnaseq
+
+import com.fulcrumgenomics.FgBioDef.{FgBioEnum, FilePath, PathPrefix, PathToBam, SafelyClosable, plotDescription}
+import com.fulcrumgenomics.bam.api._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.commons.CommonsDef._
+import com.fulcrumgenomics.commons.io.PathUtil
+import com.fulcrumgenomics.commons.util.{DelimitedDataParser, LazyLogging, SimpleCounter}
+import com.fulcrumgenomics.sopt.cmdline.ValidationException
+import com.fulcrumgenomics.sopt.{arg, clp}
+import com.fulcrumgenomics.util.{Io, Metric, ProgressLogger, Rscript}
+import enumeratum.EnumEntry
+import org.apache.commons.math3.exception.MathIllegalArgumentException
+import org.apache.commons.math3.stat.correlation.{PearsonsCorrelation, SpearmansCorrelation}
+import org.apache.commons.math3.stat.regression.SimpleRegression
+
+import scala.collection.immutable.IndexedSeq
+import scala.collection.mutable.ArrayBuffer
+import scala.io.Source
+import scala.util.Failure
+
+sealed trait ErccMixture extends EnumEntry
+object ErccMixture extends FgBioEnum[ErccMixture] {
+  def values: IndexedSeq[ErccMixture] = findValues
+  case object Mix1 extends ErccMixture
+  case object Mix2 extends ErccMixture
+}
+
+object CollectErccMetrics {
+  private[rnaseq] val StandardMixtureMetadataPath = "/com/fulcrumgenomics/rnaseq/ErccStandardMixtures.txt"
+}
+
+@clp(group = ClpGroups.RnaSeq, description=
+  """
+    |Collects metrics for ERCC spike-ins for RNA-Seq experiments.
+    |
+    |Currently calculates per-transcript ERCC metrics and summarizes dose response, but does not calculate fold-change
+    |response.
+    |
+    |The input BAM should contain reads mapped to a reference containing the ERCC transcripts.  The reference may have
+    |additional contigs, for example, when concatenating the sample's reference genome and the ERCC transcripts.  The
+    |BAM should have sequence lines in the header matching the ERCC ids (ex. ERCC-00130 or ERCC-00004).
+    |
+    |The standard ERCC transcripts, including their unique IDs and concentrations, are taken from
+    |[here](https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt).  The second column lists the ERCC
+    |transcript identifier which should be present as a sequence line in the BAM header, while columns four and five
+    |give the concentration of the transcript for mixtures #1 and #2.
+    |
+    |The choice of mixture to use can be specified with the `--mixture-name` option (either `Mix1` or `Mix2`), or with
+    |a file containing a custom list of transcripts and concentrations using the `--custom-mixture` option as follows.
+    |The custom mixture file should be tab-delimited file containing the following columns:
```

Comment:
Yeah, `DelimitedDataParser` requires a header in the file (which is silly) but I am not going to address that here.